test/OrchardCore.Tests/Modules/OrchardCore.Tenants/Services/TenantValidatorTests.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
+using Moq;
+using OrchardCore.Data;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Models;
+using OrchardCore.Tenants.Services;
+using OrchardCore.Tenants.ViewModels;
+using Xunit;
+
+namespace OrchardCore.Modules.Tenants.Services.Tests
+{
+    public class TenantValidatorTests
+    {
+        private readonly IList<ShellSettings> _shellSettings = new List<ShellSettings>();
+
+        public TenantValidatorTests() => SeedTenants();
+
+        [Theory]
+        [InlineData("Tenant1", "tenant1", "", "Feature Profile", new[] { "A tenant with the same name already exists." })]
+        [InlineData("Tenant5", "tenant3", "", "Feature Profile", new[] { "A tenant with the same host and prefix already exists." })]
+        [InlineData("Tenant5", "tenant3", null, "Feature Profile", new[] { "A tenant with the same host and prefix already exists." })]
+        [InlineData("Tenant5", "", "example2.com", "Feature Profile", new[] { "A tenant with the same host and prefix already exists." })]
+        [InlineData("Tenant6", "tenant4", "example4.com", "Feature Profile", new[] { "A tenant with the same host and prefix already exists." })]
+        [InlineData("", "tenant7", "example1.com", "Feature Profile", new[] { "The tenant name is mandatory." })]
+        [InlineData("Tenant7", "tenant7", "", "Feature", new[] { "The feature profile does not exist." })]
+        [InlineData("@Invalid Tenant", "tenant7", "example1.com", "Feature Profile", new[] { "Invalid tenant name. Must contain characters only and no spaces." })]
+        [InlineData("Tenant7", null, "  ", "Feature Profile", new[] { "Host and url prefix can not be empty at the same time." })]
+        [InlineData("Tenant7", "/tenant7", "", "Feature Profile", new[] { "The url prefix can not contain more than one segment." })]
+        [InlineData("@Invalid Tenant", "/tenant7", "", "Feature Profile", new[] { "Invalid tenant name. Must contain characters only and no spaces.", "The url prefix can not contain more than one segment." })]
+        [InlineData("Tenant8", "tenant8", "example4.com", "Feature Profile", new[] { "A tenant with the same name already exists." })]
```

Comment:
@Piedone I just added the new case that you posted to make it clear that's a another use case which which we don't think about it before. The final change I will do should fix both combinations:

1. A subset of B
2. A superset of B